### ISSUE TITLE
Coinbase Buy Widget Functionality?

### ISSUE THREAD

**Issue by no1youknowz** [OpenBazaar/OpenBazaar-Server#451]
To further facilitate the reach of OB.  I'd love to see the barrier to bitcoin lowered immensely.

Here is the link to the Coinbase Buy Widget:  https://developers.coinbase.com/docs/buy-widget

Having something like a buy bitcoin to transfer to the seller button.  Would allow someone like me to market on Facebook, knowing that probably 99% of visitors don't know what bitcoin is.  But then exposing them to OB and bitcoin and leaving them finding out what bitcoin is.  Probably further transactions would be done in bitcoin.

Probably this is a long ways off, but having a feature like this on the roadmap would be great!

Thanks

**Comment by ABISprotocol** [comment]
I do not recommend this due to the privacy problems inherent in Coinbase.

I would also note that in addition to trying to warn users of the dangers of web wallets (something I have been doing for some time), I have just decided after discussion with bitcoin.org maintainers - who are not intending to materially alter criteria which have been used for a very long time to analyze wallets that appear on that site - that I will no longer participate in discussions of issues at bitcoin.org and that there is a need for **mobile, desktop, and hardware wallets to be assessed with particular caution by potential users of bitcoin.org due to the absence of proper criteria at bitcoin.org for assessment for risk in circumstances where a wallet is expressly designed as a portal to a bank or a banking network, or as a portal to an exchange or brokerage.**

Please see: https://github.com/bitcoin-dot-org/bitcoin.org/pull/1223#issuecomment-230637124

See also discussion at https://github.com/bitcoin-dot-org/bitcoin.org/issues/996

Additionally, Coinbase is a brokerage - not a wallet that allows users control over their funds.

See: https://github.com/btcfoundationedcom/btcfoundationedcom.github.io/blob/master/guides/en/buying.md for reference.

Coinbase has stated it is evolving into what it calls a hybrid wallet-exchange.  
See: https://medium.com/the-coinbase-blog/coinbase-is-not-a-wallet-b5b9293ca0e7

The user is not in control of these funds and the risks of these models are not properly assessed.

I oppose a Coinbase Buy Widget being included in OB.

**Comment by no1youknowz** [comment]
Thanks for that @ABISprotocol, very interesting.

What do you suggest instead to have similar capability so that the masses can spend using bitcoin?

**Comment by ABISprotocol** [comment]
This is a tough question you've asked, @no1youknowz ~  Having some kind of alternative with similar capability would certainly be good.  I have concerns about the use of Shapeshift due to its [terms of service](https://info.shapeshift.io/sites/default/files/ShapeShift_Terms_Conditions%20v1.1.pdf) - see Privacy and Transparency Statement at end of ToS - OB has evaluated using Shapeshift in the past, but I don't know if it is considered viable or not at present.

Currently:

OpenBazaar (is designed to support) [four payment options](https://docs.openbazaar.org/03.-OpenBazaar-Protocol/#32-trade-protocol) with Bitcoin:

    _Direct payment_
        Alice transfers bitcoin to Bob in a simple payment
    _Double-deposit 'MAD' escrow_
        Alice and Bob create a 2-of-2 multisignature escrow address
        Alice and Bob need to agree to release funds from that address
    _Notary escrow_
        Alice, Bob, and Charlie (a third party 'notary') create a 2-of-3 multisignature escrow address
        Two members of this party are required to release funds from the address
    _Notary pooled escrow_
        Alice, Bob, and others (multiple third party notaries) create a n-of-m multisignature escrow address
        Multiple parties from the pool of notaries are required to release funds from the address

To answer your question though, perhaps a similar capability would be to have an OB category that would direct buyers specifically to an area where they would see only sellers of cryptocurrency, and a button which would enable the user to specify prices at which they would be willing to buy.  

Right now it is difficult to find what you would like to buy on OB. (See for example, [this](https://duosear.ch/))

Alternatively, a page could simply show buttons which display prices of cryptocurrency from different sellers and the potential buyers could select which one (which seller / which price) is more favorable to them at that time.  Such a functionality if added would make OB sellers more able to offer bitcoin or other cryptocurrencies for sale with much greater ease.  

Perhaps related to this:  An issue in the deprecated version explored altcoin support (you might not be able to see this unless you are dev) https://github.com/OpenBazaar/OpenBazaar/issues/954

In January 2015, I stated that "I want openbazaar to work w/ MyceliumCom (Local Trader aspects) & Mycelium Straight (minus KYC/AML)" 
The response I got back from MyceliumCom on twitter was "Adding @openbazaar will take some time, plus it needs to mature more. Straight we'll definitely have soon"

Distributed currency exchange on OB is covered in more significant detail here:
https://gist.github.com/drwasho/aa6ab79e92f2a876073e
It deals with crypto-crypto, crypto-fiat, and fiat-fiat sort of trades.  It is a long read, grab some popcorn and settle in.

Technically, using such methods as a Duosearch link -- https://duosear.ch/ -- you can right now show someone on Facebook (as you suggested) a link to a listing, even someone who has never used OB.  However, they need to actually download and use OB in order to make a purchase at present.

**Comment by tmmunro** [comment]
"However, they need to actually download and use OB in order to make a purchase at present." -- or install the mobile app "Bazaar Hound" to make the purchase

**Comment by ABISprotocol** [comment]
Good point, @tmmunro ~ thank you.  (I'm editing my comment of July 6, 2016 above to reflect that.)
